be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -1376,7 +1376,7 @@ Status SegmentIterator::_lookup_ordinal_from_pk_index(const RowCursor& key, bool
     // when is_include is false, we shoudle append KEY_NORMAL_MARKER to the
     // encode key. Otherwise, we will get an incorrect upper bound.
     encode_key_with_padding<RowCursor, true>(
-            &index_key, key, _segment->_tablet_schema->num_key_columns(), is_include, true);
+            &index_key, key, _segment->_tablet_schema->num_key_columns(), is_include, false);
```

Comment:
https://github.com/apache/doris/pull/21770 set it to true, this pr set it to false, which one is correct? And why?